Basic-Car-Maintenance/Shared/Onboarding/Views/WelcomeViewAddVehicle.swift

```diff
@@ -9,99 +9,177 @@ import SwiftUI
 
 struct WelcomeViewAddVehicle: View {
     
-    // Logic to remember Onboarding screen to not load again when app is launched
-//    @AppStorage("isFirstTime") private var isFirstTime: Bool = true
+    // Logic to remember OnboardingScreen to not load again when app is launched
+    @AppStorage("isFirstTime") private var isFirstTime: Bool = true
+    @AppStorage("vehicleName") private var vehicleName: String = ""
+    @AppStorage("vehicleMake") private var vehicleMake: String = ""
+    @AppStorage("vehicleModel") private var vehicleModel: String = ""
     @Environment(\.dismiss) var dismiss
-    
-    @State private var vehicleName: String = ""
-    @State private var vehicleMake: String = ""
-    @State private var vehicleModel: String = ""
+    @State private var validationAlertName: Bool = false
+    @State private var validationAlertMake: Bool = false
+    @State private var validationAlertModel: Bool = false
+    @State private var onboardingViewCompletedAlert: Bool = false
     
     var body: some View {
         VStack(spacing: 15) {
             VStack {
-                Text("Add the details below")
-                HStack(spacing: 5) {
-                    Text("about")
-                    Text("your vehicle")
-                        .foregroundStyle(Color("basicGreen"))
-                }
+                headerView
+                vehicleDetailsView
+                bottomText
+                Spacer(minLength: 10)
+                addVehicleButton
             }
-            .font(.largeTitle.bold())
-            .multilineTextAlignment(.center)
-            .padding(.top, 65)
-            .padding(.bottom, 15)
-            
-            VStack {
-                Image(systemName: "car.side.lock.open")
-                    .font(.system(size: 45))
+        }
+        .frame(maxWidth: .infinity, maxHeight: .infinity)
+        .background(Color(UIColor.secondarySystemBackground).ignoresSafeArea())
+        .navigationBarBackButtonHidden(true)
+        .toolbar { backButton }
+    }
+    
+    private var headerView: some View {
+        VStack {
+            Text("Add the details below")
+            HStack(spacing: 5) {
+                Text("about")
+                Text("your vehicle")
                     .foregroundStyle(Color("basicGreen"))
```

Comment:
this doesn't need to be in an HStack to get the color it could be:
```
Text("Add the details below about ") +
            Text("your vehicle")
                .foregroundStyle(Color("basicGreen"))
```

Reply:
Ok, I will do it 👍